File: generate_wallets.py
import secrets

print("=========================================")
print("   🚀 GENERATING 10 HOT WALLETS 🚀       ")
print("=========================================\n")

env_output = "\n# --- BOT WALLET FLEET ---\n"
private_keys = []

for i in range(1, 11):
    # An Ethereum private key is just 32 bytes of secure random entropy!
    # No external libraries needed.
    priv_key = secrets.token_hex(32)
    private_keys.append((i, priv_key))
    env_output += f"BOT_WALLET_{i}={priv_key}\n"

print("✅ Private Keys generated successfully!\n")

print("1️⃣  Paste these private keys into MetaMask (or Rabby) using 'Import Account':")
print("     (MetaMask will calculate the public 0x... address for you automatically)")
print("-----------------------------------------")
for i, key in private_keys:
    print(f"Wallet {i:2d} Private Key: {key}")

print("\n\n2️⃣  Once you fund them with $5 each, paste this exact block into engine/.env:")
print("-----------------------------------------")
print(env_output)
print("=========================================\n")

File: generate_wallets.js
const { ethers } = require("ethers");
const fs = require("fs");

console.log("=========================================");
console.log("   🚀 GENERATING 10 HOT WALLETS 🚀       ");
console.log("=========================================\n");

let envOutput = "\n# --- BOT WALLET FLEET ---\n";
let addressList = [];

for (let i = 1; i <= 10; i++) {
    const wallet = ethers.Wallet.createRandom();

    // Store for console display
    addressList.push({ id: i, address: wallet.address });

    // Format for rust engine .env (strip the 0x prefix from private key)
    envOutput += `BOT_WALLET_${i}=${wallet.privateKey.slice(2)}\n`;
}

console.log("✅ Wallets generated successfully!\n");

console.log("1️⃣  Copy these addresses and send $5 of ETH on Arbitrum to each:");
console.log("-----------------------------------------");
addressList.forEach(w => {
    console.log(`Wallet ${w.id.toString().padStart(2, ' ')}: ${w.address}`);
});

console.log("\n\n2️⃣  Paste this exact block into your engine/.env file:");
console.log("-----------------------------------------");
console.log(envOutput);
console.log("=========================================\n");
